### PR TITLE
collate_results fix

### PR DESCRIPTION
**Pull request by mattssca**
# Pull Request Checklists

**Important:** When opening a pull request, keep only the applicable checklist and delete all other sections.

### Required

- [x] I tested the new code for my use case (please provide a reproducible example of how you tested the new functionality)

- [x] I ensured all dplyr functions that commonly conflict with other packages are fully qualified. 

This can be checked and addressed by running `check_functions.pl` and responding to the prompts. Test your code _after_ you do this.

- [x] I generated the documentation and checked for errors relating to the new function (e.g. `devtools::document()`) and added `NAMESPACE` and all other modified files in the root directory and under `man`. 

### Optional but preferred with PRs

- [ ] I updated and/or successfully knitted a vignette that relies on the modified code (which ones?)

### Required

- [x] I documented my function using [ROxygen style](https://jozef.io/r102-addin-roxytags/#:~:text=Inserting%20a%20skeleton%20%2D%20Do%20this,Shift%2BAlt%2BR%20).)

- [x] All parameters for the function are described in the documentation and the function has a decriptive title. 

Example:
```
#' Use GISTIC2.0 scores output to reproduce maftools::chromoplot with more flexibility
#'
#' @param scores output file scores.gistic from the run of GISTIC2.0
#' @param genes_to_label optional. Provide a data frame of genes to label (if mutated). The first 3 columns must contain chromosome, start, and end coordinates. Another required column must contain gene names and be named `gene`. (truncated for example)
#' @param cutoff optional. Used to determine which regions to color as aberrant. Must be float in the range [0-1]. (truncated for example)
```

- [ ] My function uses a library that isn't already a dependency of GAMBLR and I made the package aware of this dependency using the function documentation `import` statment. 

Example:
```
#' @return nothing
#' @export
#' @import tidyverse ggrepel
```

## Checklist for changes to existing code

- [ ] I added/removed arguments to a function and updated documentation for all changed/new arguments

- [ ] I tested the new code for compatability with existing functionality in the Master branch (please provide a reprex of how you tested the original functionality)

### CODE REVIEW

File: R/utilities.R
Comment (rdmorin) [inline R/utilities.R:1323]: This is a partial fix but how will this work when use_cache = TRUE?

Comment (rdmorin) [reply R/utilities.R:1323]: I know you said it's only fixing one of the two modes but this code change is guaranteed to not allow the other mode to ever work due to the scope of the variables.

**Review by rdmorin — CHANGES_REQUESTED** [review]
The solution is guaranteed to never work for the other (more common) use case. Please adjust the scope of the variables that are required by both modes to address this.

File: R/utilities.R
Comment (mattssca) [reply R/utilities.R:1323]: So I guess both options (from_cache = TRUE/FALSE) would work if they have access to the same output_files? i.e moving the output_file code outside the from_cache if statement. Allowing from_cache = FALSE to find the file if write_to_file = TRUE. Is this correct?

```
 output_file = config::get("results_merged")$collated
 output_base = config::get("project_base")
 output_file = paste0(output_base, output_file)
 output_file = glue::glue(output_file)
```

**Comment by rdmorin** [comment]
Please don't submit a PR until the problem is fully fixed. Use a draft PR instead if you want feedback on the partial fix. I also would like to see working examples that test both modes of from_cache in the PR.

**Comment by mattssca** [comment]
I have now marked this PR as a draft, I will update and provide the information you requested once it's ready for review. Thanks!

File: R/utilities.R
Comment (rdmorin) [inline R/utilities.R:1324]: Adam after you left we discussed this further and opted to save the results to that `results/shared` folder so please change this line to 
```
output_base = config::get("project_base")
```

Comment (rdmorin) [reply R/utilities.R:1324]: In this case those both point to the same file because every gambl clone has a symlink `results` that points to `results_local`. You should familiarize yourself with the layout and organization of the gambl repo especially how we use symlinks. GAMBLR should be using the first path in this case because it's writing to `results` rather than one of the paths in the repo that is tracked in git.

Comment (mattssca) [reply R/utilities.R:1324]: Perfect, Thanks for clarifying, makes 100% sense.

**Comment by mattssca** [comment]
This function has now been updated with appropriate file paths for reading and exporting cached results. The function has also been getting a facelift in terms of expanded in-line comments, parameter descriptions and extended examples. 

When testing this function I was not allowed to write to file:  `Error: Cannot open file for writing`

However, I tested this function with `from_cached = FALSE` and `write_to_file = FALSE` for both `genome` and `capture` with success. As such:

```
genome_collated = collate_results(seq_type_filter = "genome", from_cache = FALSE, write_to_file = FALSE)
capture_collated = collate_results(seq_type_filter = "capture", from_cache = FALSE, write_to_file = FALSE)
```
I also tested using `these_samples_metadata` and `join_with_full_metadata`:

```
fl_metadata = get_gambl_metadata(seq_type_filter = "genome") %>% dplyr::filter(pathology == "FL")
fl_collated = collate_results(seq_type_filter = "genome", join_with_full_metadata = TRUE, these_samples_metadata = fl_metadata, write_to_file = FALSE, from_cache = FALSE)

```
Lastly, I tested this function with using the sample_table parameter as well:
```
fl_samples = get_gambl_metadata(seq_type_filter = "genome") %>% dplyr::filter(pathology == "FL") %>% dplyr::select(sample_id, patient_id, biopsy_id)
fl_collated = collate_results(sample_table = fl_samples, seq_type_filter = "genome", from_cache = FALSE, write_to_file = FALSE)
```

File: R/utilities.R
Comment (rdmorin) [inline R/utilities.R:1383]: Out of interest, why was this change made?

Comment (mattssca) [reply R/utilities.R:1383]: I reverted a change that I made in [this](https://github.com/morinlab/GAMBLR/pull/133/commits/213e9f8f0e3721fefbf3e0de9164173f96d0adfd) commit, from when I first started looking into this function last week.

Comment (rdmorin) [reply R/utilities.R:1383]: Then i have the same question but in regards to that commit instead. Why was it changed?

Comment (mattssca) [reply R/utilities.R:1383]: I thought it should be the other way around (i.e sample_table , meta_data) but after some thinking and testing, it was correct from the beginning (i.e meta_data, sample_table).

File: R/utilities.R
Comment (rdmorin) [inline R/utilities.R:1295]: There's a typo here (parameter is misspelled)

File: R/utilities.R
Comment (rdmorin) [inline R/utilities.R:1305]: typo (cached)

File: R/utilities.R
Comment (rdmorin) [inline R/utilities.R:1306]: Presumably you meant to name your variable `capture_collated_eerything`

File: R/utilities.R
Comment (rdmorin) [inline R/utilities.R:1308]: cheched?

File: R/utilities.R
Comment (rdmorin) [inline R/utilities.R:1297]: For the sake of our future sanity can you add some details in this section that tell the user where the cached results are written?

File: R/utilities.R
Comment (rdmorin) [inline R/utilities.R:1372]: typo

**Review by rdmorin — CHANGES_REQUESTED** [review]
There are lots of aesthetic fixes that should be applied before this gets merged

**Review by rdmorin — COMMENTED** [review]
Good to go. Try to be more careful reviewing your documentation for typos before you submit a PR.

**Review by rdmorin — APPROVED** [review]

**Pull request merged**
